src/commands/lobby.ts

```diff
@@ -1,50 +1,75 @@
+import { ComponentType } from "discord.js";
 import GameManager from "../util/GameManager/GameManager.js";
 import { fetchMember } from "../util/MemberUtil/MemberFetch.js";
+import LobbyEmbed from "../util/Embeds/LobbyEmbed.js";
 
 const __lobby: InteractionHandlerPayloads.GuildChatInputCommand = {
     name: "start",
     description: "Start a Black Jack Game",
 
     async execute(interaction) {
         const gameManager = new GameManager(interaction);
+        const lobbyEmbed = new LobbyEmbed(gameManager.players);
 
-        //test add player
-        gameManager.players.addPlayer(
-            await fetchMember(interaction.guildId, interaction.member.id),
-            interaction.member
-        );
-
-        // get the lobby emebed
-        const embed = gameManager.LobbyEmbed();
-
-        // creates the join, leave and start now buttons
-        const actionRow = new ActionRowBuilder<ButtonBuilder>().addComponents(
-            new ButtonBuilder()
-                .setCustomId("join-game")
-                .setLabel("Join")
-                .setStyle(ButtonStyle.Success),
-
-            new ButtonBuilder()
-                .setCustomId("leave-game")
-                .setLabel("Leave")
-                .setStyle(ButtonStyle.Danger),
-
-            new ButtonBuilder()
-                .setCustomId("start-game")
-                .setLabel("Start Now")
-                .setStyle(ButtonStyle.Primary)
-        );
-
-        await interaction.reply({
+        console.log("A lobby has been created");
+
+        // create and display the lobby emebed
+        const { embeds: embed, components: actionRow } =
+            lobbyEmbed.CreateEmbed();
+
+        const message = await interaction.reply({
             embeds: [embed],
             components: [actionRow],
         });
 
+        const collector = message.createMessageComponentCollector({
+            filter: (i) => i.user.id === interaction.user.id,
+            componentType: ComponentType.Button,
+            time: 120_000,
+        });
+
+        // collector that responds to the buttons
+        collector.on("collect", async (i) => {
+            if (i.customId === "join-game") {
+                // add player if not already in the lobby
+                if (!gameManager.players.getPlayer(interaction.member.id)) {
+                    gameManager.players.addPlayer(
+                        await fetchMember(
+                            interaction.guildId,
+                            interaction.member.id
+                        ),
+                        interaction.member
+                    );
+                    // update lobby player list
+                    const { embeds: updatePlayers } = lobbyEmbed.CreateEmbed();
+                    interaction.editReply({ embeds: [updatePlayers] });
+                }
+                console.log(`Added player ${interaction.member.displayName}`);
+                await i.deferUpdate();
```

Comment:
Delete line 48.
deferUpdate() should occur first and does not need to be awaited as we do not update the embed via .update() on the i interaction, but on the original interaction.